financial_calculations/forward_curves.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+import pandas as pd
+from datetime import datetime, timedelta
+from dateutil.relativedelta import relativedelta
+from scipy.optimize import minimize
+from utils import (
+    discount_cash_flows
+)
+from financial_calculations.bond_cash_flows import (
+    create_semi_bond_cash_flows
+)
+
+def bootstrap_forward_curve(cmt_data, market_close_date, par_value, initial_guess=0.04):
+    """
+    Bootstrap the spot curve using bond data.
+    
+    Parameters:
+    -----------
+    cmt_data : list of tuples
+        Each tuple contains (maturity_years, coupon_rate) for bonds.
+        Coupon rate should be expressed as a decimal (e.g., 5% as 0.05).
+    market_close_date : datetime or datetime64[D]
+        The market close date from which the cash flows are discounted.
```

Comment:
"to which" instead of "from which".  future cash flows are discounted to the market close date.